### PR TITLE
Make Flutter Driver actively wait for runnable isolate

### PR DESCRIPTION
**Pull request by jakubfijalkowski**
This PR changes the way Flutter Driver waits for runnable `main` isolate. Previously, it tried to get the info about isolate again if the isolate was in `None` state when first getting it. Unfortunately, this seems to be overly optimistic in CI environments like GitHub Actions. [From our experience](https://github.com/leancodepl/patrol/issues/272) it seems that the main isolate running on an emulator on a macOS VM might take a little bit longer to become runnable. If it stays in `None`, it will miss being resumed.

Now, the client is polled periodically until the isolate is in other than `None` state. Then it proceeds to resume the isolate if it is paused.

Why polling instead of waiting for an event? Because it is hard to detect when isolate changes `pauseEvent` based on the events in the `onIsolateEvent` stream. We don't have a list of run-related events (that I know of) and it would require hardcoding them, which would result in this problem resurfacing after another pause-related event is added. The other option would be to `getIsolate` on every event, which seems to be quite a heavy solution (as there is a spike of events at the beginning). Polling every 100ms seems like the least heavy option.

There is another change that adds `PausePostRequest` to a `pauseEvent` that results in `resume`s - this looks like an omission.

This issue fixes problem that occurs in [Patrol](https://github.com/leancodepl/patrol/issues/272).
It should also somewhat help with this issue: https://github.com/flutter/flutter/issues/50898 .

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by pdblasi-google — APPROVED** [review]
LGTM!

File: packages/flutter_driver/test/src/real_tests/flutter_driver_test.dart [hunk 1]
```diff
@@ -1055,6 +1090,15 @@ vms.Response? makeFakeResponse(
   });
 }
 
+void Function(vms.Isolate) changeIsolateEventAfter(int gets, vms.Event nextEvent) {
+  return (vms.Isolate i) {
+    gets--;
```
Comment (dnfield) [inline packages/flutter_driver/test/src/real_tests/flutter_driver_test.dart:1095]: nit: style guide for Flutter repo says to use `-=`.

**Review by dnfield — APPROVED** [review]

File: packages/flutter_driver/test/src/real_tests/flutter_driver_test.dart (hunk 1 above)
Comment (jakubfijalkowski) [reply packages/flutter_driver/test/src/real_tests/flutter_driver_test.dart:1095]: Sorry, must have missed that. Will fix (in a couple of days if that's not a problem)!

Comment (dnfield) [reply packages/flutter_driver/test/src/real_tests/flutter_driver_test.dart:1095]: No worries

**Comment by goderbauer** [comment]
(Triage) @jakubfijalkowski Can you rebase this PR to the latest master to make the "ci.yaml validation" check happy?

**Comment by jakubfijalkowski** [comment]
@goderbauer done!

**Comment by goderbauer** [comment]
@jakubfijalkowski Can you rebase this one more time to trigger the Google testing check again? Sorry about the inconvenience.

**Comment by jakubfijalkowski** [comment]
@goderbauer no worries, done!

**Pull request merged**